Hunting Queries/SecurityEvent/FirewallRuleChanges_using_netsh.yaml

```diff
@@ -0,0 +1,30 @@
+id: 3dc5dc8b-160b-407e-9925-24a91e3599df
+name: Firewall rule changes using netsh
+description: |
+  This query will show firewall rule changes using netsh utility.
+  This technique was seen in relation to Solarigate attack but the results can indicate potential  malicious activity used in different attacks.
+  Refer to netsh syntax: https://docs.microsoft.com/windows-server/administration/windows-commands/netsh
+severity: Low
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent
+tactics:
+  - Execution
+relevantTechniques:
+  - T1204
+query: |
+  let timeframe = 1d;
+  let AccountAllowList = dynamic(['SYSTEM']);
+  let tokens = dynamic(["add", "delete", "set"]); 
+  SecurityEvent
+  | where TimeGenerated >= ago(timeframe)
+  | where Process =~ "netsh.exe" 
+  | where CommandLine has_all ("advfirewall", "firewall") and CommandLine has_any (tokens)
+  | where AccountType != "Machine" and Account !in (AccountAllowList)
+  | extend KeyValuePairs = extract_all(@"(?P<key>\w+)=(?P<value>\S+)", dynamic(["key","value"]), CommandLine)
+  | mv-apply KeyValuePairs on (
+      summarize CommandLineParsed = make_bag(pack(tostring(KeyValuePairs[0]), KeyValuePairs[1]))
+  )
+  | summarize count() , StartTime= min(TimeGenerated), EndTime=max(TimeGenerated) by Computer, Account, SubjectDomainName, SubjectUserName, CommandLine, Process, ParentProcessName, tostring(CommandLineParsed)
```

Comment:
Order of operation is not an issue here, but readability is usually better if this is written as -

`| summarize StartTime= min(TimeGenerated), EndTime=max(TimeGenerated), count() by `